python/lsst/meas/base/plugins.py

```diff
@@ -198,8 +198,12 @@ def measure(self, measRecord, exposure):
         # Compute the median variance value for each pixel not excluded by the mask and write the record.
         # Numpy median is used here instead of afw.math makeStatistics because of an issue with data types
         # being passed into the C++ layer (DM-2379).
-        medVar = numpy.median(pixels.getVarianceArray()[logicalMask])
-        measRecord.set(self.varValue, medVar)
+        if numpy.sum(logicalMask) > 0:
+            medVar = numpy.median(pixels.getVarianceArray()[logicalMask])
+            measRecord.set(self.varValue, medVar)
+        else:
+            measRecord.set(self.varValue, numpy.NaN)
+            raise bl.MeasurementError("Footprint empty, or all pixels are masked, can't compute median", bl.FlagHandler.FAILURE)
```

Comment:
Would it be appropriate to define a specific flag for this failure mode?

Is the reference to `bl.FlagHandler.FAILURE` appropriate? In some sense, I guess it is, but I think if we were doing it properly we'd actually have to define `FAILURE_BAD_CENTROID` above as `bl.FlagHandler.FAILURE+1` rather than just assuming it's 1; if we _can_ assume it's 1, then maybe we can just assume this is 0. Maybe a specific reference to `FlagHandler.FAILURE` is helpful to the reader? (On the other hand, I just wasted 5 minutes wondering about it...)